Odontograma

NadiaArroyo25 opened the pull request.

NadiaArroyo25 committed:
prueba con notas

NadiaArroyo25 committed:
se muestra en new

aqui podemos ver la imagen en new

NadiaArroyo25 committed:
punto dos

aqui no me muestra la imagen

NadiaArroyo25 committed:
Update _medical_record.html.erb

davepcage commented on app/controllers/medical_records_controller.rb:
```diff
@@ -13,7 +13,13 @@ def show
+    @medical_record.build_image_attachment
+
+    if params[:medical_record] && params[:medical_record][:image]
+    @medical_record.image.attach(params[:medical_record][:image])
```
Indentation
```suggestion
      @medical_record.image.attach(params[:medical_record][:image])
```

davepcage commented on app/controllers/medical_records_controller.rb:
```diff
@@ -13,7 +13,13 @@ def show
+    @medical_record.image.attach(params[:medical_record][:image])
+    end
   end
+  
```
You don´t need this line
```suggestion
```

davepcage commented on app/controllers/medical_records_controller.rb:
```diff
@@ -22,6 +28,8 @@ def edit
   def create
     @medical_record = MedicalRecord.new(medical_record_params)
+    @medical_record.image.attach(params[:medical_record][:image]) if params[:medical_record][:image]
+ 
```
You don´t need this line
```suggestion
```

davepcage commented on app/controllers/medical_records_controller.rb:
```diff
@@ -37,6 +45,7 @@ def create
   # PATCH/PUT /medical_records/1 or /medical_records/1.json
   def update
     respond_to do |format|
+      
```
You don´t need this line
```suggestion
```

davepcage commented on app/models/medical_record.rb:
```diff
@@ -1,3 +1,4 @@
   belongs_to :patient
-end
+  has_one_attached :image
+end
```
the last line must be empty
```suggestion
end

```

davepcage commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
     <%= form.submit %>
   </div>
+  <% if @medical_record.image.attached? %>
+  <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
```
Identation
```suggestion
    <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
```

AlfredoJA replied:
If you want to display the recorded image, you need to use 
```suggestion
  <%= image_tag(@medical_record.image, alt: "odontograma", class:"d-block w-100") %>
```
In order to not display always the same image.
But I'm not sure understand what you want to do.

davepcage commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
   </div>
+  <% if @medical_record.image.attached? %>
+  <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
+
```
Delete this line
```suggestion
```

davepcage commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
+  <% if @medical_record.image.attached? %>
+  <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
+
+  <!-- Mensaje o marcador de posición si no hay imagen asociada -->
```
What is the purpose of this comment?

AlfredoJA replied:
Try to add english comments too.

davepcage commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
+  <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
+
+  <!-- Mensaje o marcador de posición si no hay imagen asociada -->
+  <% end%>
```
Space
```suggestion
  <% end %>
```

davepcage commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
+  <% end%>
 <% end %>
+
+
```
Only the last empty line
```suggestion
```

davepcage commented on app/views/medical_records/_medical_record.html.erb:
In line 6 you have opened with "< td >" and closed with "< /th >".
Fix it, please

NadiaArroyo25 replied:
ups!

davepcage commented on db/migrate/20240418224031_add_image_to_medical_records.rb:
```diff
@@ -0,0 +1,4 @@
+class AddImageToMedicalRecords < ActiveRecord::Migration[7.1]
+  def change
```
Why have you created this migration?

AlfredoJA replied:
@davepcage I think a better english is "Why did you create this migration?"

AlfredoJA replied:
You don't need a new migration for "image", because you are using ActiveRecordStorage table, so you can delete this file.
But first remember do a rollback of your database, you can use:
`rails db:migrate:status`
to know what migrations was runned,
then you can use:
`rails db:rollback STEP=n`
where `n` is the number of rollback you want to do.
for example:
`rails db:migrate:status` returns:
```
   up     20240411165333  Another
   up     20240411184039  AddImageToMedicalRecords
   up     20240418010041  AddNotesToMedicalRecords
```
then you can do:
`rails db:rollback STEP=2`
to rollback until 20240411184039
after rollback if you runs `rails db:migrate:status` again it should shows this:
```
   up       20240411165333  Another
   down     20240411184039  AddImageToMedicalRecords
   down     20240418010041  AddNotesToMedicalRecords
```
and when you confirm the migration is down you can delete the file, in order to not have issues in the future.

NadiaArroyo25 replied:
Thanks!

davepcage commented on app/views/medical_records/_medical_record.html.erb:
```diff
@@ -13,6 +13,14 @@
       <th scope="row">Esta de salud:</th>
       <td><%=medical_record.current_state %></td>
     </tr>
+    <tr>
+      <th scope="row">Notas</th>
+      <td><%=medical_record.notes %></td>
+    </tr>
+    <tr>
+      <th scope="row">Imagen</th>
+      <td><%=image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100")%></td>
+    </tr>
```
convention for table:
https://getbootstrap.com/docs/4.6/content/tables/

```suggestion
<table class="table">
  <thead class="thead-dark">
    <tr>
      <th scope="col">First</th>
      <th scope="col">Last</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Mark</td>
      <td>Otto</td>
    </tr>
    <tr>
      <td>Jacob</td>
      <td>Thornton</td>
    </tr>
  </tbody>
</table>
```

AlfredoJA replied:
Her format is ok @davepcage, maybe she can improve it, but the use is ok, if She don't want to add headers.

NadiaArroyo25 replied:
<img width="1216" alt="Captura de pantalla 2024-04-21 a la(s) 4 42 39 p m" src="https://github.com/NadiaArroyo25/prueba1/assets/130617270/ff9fa8a2-a3e1-4d7f-983b-765934a9abde">
I put it like this because I want only two rows and one black row

AlfredoJA commented on app/controllers/medical_records_controller.rb:
```diff
@@ -22,6 +28,8 @@ def edit
   # POST /medical_records or /medical_records.json
   def create
     @medical_record = MedicalRecord.new(medical_record_params)
+    @medical_record.image.attach(params[:medical_record][:image]) if params[:medical_record][:image]
```
I think line 30 is enough to attach the image, as you can see in this example https://guides.rubyonrails.org/active_storage_overview.html#has-many-attached.
Given this is a single record you don't need to user `images: []` as the example shown, only `:image`.

AlfredoJA commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,19 @@
   </div>
 
+  <div>
+  <%= form.label :notes, style: "display: block" %>
```
```suggestion
    <%= form.label :notes, style: "display: block" %>
```

Ligator commented:
@NadiaArroyo25, can you please explain why did you need to add ActiveStorage?
ActiveStorage allows you and your users to upload images, for example, when a doctor needs to upload a PDF document for a specific patient, or when a patient needs to attach photos to his/her appointment to provide more information.
I have the feeling all you wanted to do is to show your app/assets/images/Odontogram.jpg image, you don't need ActiveStorage for that because the image is already in the code.
Please confirm what happened here.

NadiaArroyo25 committed:
indentation

NadiaArroyo25 committed:
Update app/controllers/medical_records_controller.rb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/views/medical_records/_form.html.erb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/models/medical_record.rb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/views/medical_records/_form.html.erb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/views/medical_records/_form.html.erb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/views/medical_records/_form.html.erb

Co-authored-by: davepcage <71347825+davepcage@users.noreply.github.com>

NadiaArroyo25 committed:
Update app/views/medical_records/_form.html.erb

Co-authored-by: AlfredoJA <alfre.jangeles@hotmail.com>

Ligator commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +31,16 @@
   <div>
     <%= form.submit %>
   </div>
+  <% if @medical_record.image.attached? %>
+    <%= image_tag("Odontogram.jpg", alt: "odontograma", class:"d-block w-100") %>
+  <% end %>
```
You don't need to have an image attached to the medical record to show an image that is part of your repository.

NadiaArroyo25 committed:
correciones

elimine un label sin querer y no me di cuenta asi que mi código no funcionaba tambien modifique el como mostrar la imagen sin active storage, y en que vistas queria que se mostrara

NadiaArroyo25 committed:
Merge branch 'odontograma' of https://github.com/NadiaArroyo25/prueba1 into odontograma

Ligator commented on app/views/medical_records/_medical_record.html.erb:
```diff
@@ -1,10 +1,13 @@
 <div id="<%= dom_id medical_record %>">
 <table class="table table-striped">
-  
+<thead>
+
+</thead>
+
```
Remove useless lines of code.

NadiaArroyo25 replied:
listo

NadiaArroyo25 committed:
spaces

Ligator commented on app/views/medical_records/_form.html.erb:
```diff
@@ -1,19 +1,20 @@
       </ul>
     </div>
   <% end %>
-
   <div>
     <%= form.label :patient_id, style: "display: block" %>
-    <%= form.text_field :patient_id %>
+    <p><%= medical_record.patient_id %></p>
+    <%= form.hidden_field :patient_id %>
   </div>
```
You cannot close a tag after its parent tag.

Bad:
```
<div>
  <form>
</div>
 </form>
```

Good
```
<div>
  <form>
  </form>
</div>
```

NadiaArroyo25 replied:
entendido

Ligator commented on app/controllers/medical_records_controller.rb:
```diff
@@ -4,15 +4,19 @@ class MedicalRecordsController < ApplicationController
   # GET /medical_records or /medical_records.json
   def index
     @medical_records = MedicalRecord.where(patient_id: params[:patient_id]).all
+    @medical_records.each do |medical_record|
+    medical_record.image_location = "Odontogram.jpg"
+    end
```
Indentation
```suggestion
    @medical_records.each do |medical_record|
      medical_record.image_location = "Odontogram.jpg"
    end
```
However, you don't need these lines, I explain it in my next comment.

Ligator commented on app/views/medical_records/_form.html.erb:
```diff
@@ -31,7 +34,19 @@
+  </div>
+
+  <div>
+    <% if medical_record.image_location.present? %>
+      <%= image_tag(medical_record.image_location, class: "d-block w-100") %>
+    <% end %>
```
I think the way you included the image is pretty clever. However, you don't need to attach the image to the medical record at all.
You can do here something like this:
```suggestion
    <%= image_tag("Odontogram.jpg", class: "d-block w-100") %>
```
And remove the `image_location` attribute.

NadiaArroyo25 replied:
You're right, I accidentally broke something and I didn't realize what it was, so I attributed it to the image, so I made a thousand fixes, I guess they were unnecessary because it works that simple.

NadiaArroyo25 committed:
recommendations image

Ligator commented on app/views/medical_records/_medical_record.html.erb:
```diff
@@ -13,9 +12,14 @@
     </tr>
+    <tr>
+      <th scope="row">Notas</th>
+      <td><%=medical_record.notes %></td>
```
```suggestion
      <td><%= medical_record.notes %></td>
```

Ligator commented on app/views/medical_records/_form.html.erb:
```diff
@@ -1,19 +1,24 @@
   <div>
     <%= form.label :patient_id, style: "display: block" %>
-    <%= form.text_field :patient_id %>
+    <p>
+     <%= medical_record.patient_id %>
+    </p>
```
```suggestion
    <p>
      <%= medical_record.patient_id %>
    </p>
```

Ligator replied:
```suggestion
    <p>
      <%= medical_record.patient_id %>
    </p>
```

Ligator commented on app/models/medical_record.rb:
```diff
@@ -1,3 +1,4 @@
 class MedicalRecord < ApplicationRecord
   belongs_to :patient
+  attr_accessor :image_location
```
You no longer need this.

NadiaArroyo25 replied:
oh! It's true!

Ligator commented on app/controllers/medical_records_controller.rb:
```diff
@@ -37,7 +36,7 @@ def create
   def update
     respond_to do |format|
-      if @medical_record.update(medical_record_params)
+      if @medical_record.update(medical_record_params.except(:image_location))
```
You no longer need to do this exception.

Ligator commented on app/controllers/medical_records_controller.rb:
```diff
@@ -8,12 +8,12 @@ def index
   def new
     @medical_record = MedicalRecord.new(patient_id: params[:patient_id])
-  end
+   end
```
```suggestion
  end
```

Ligator commented on app/views/medical_records/_form.html.erb:
The "form" tag must be closed inside the same tag where it was opened.

Ligator commented on app/views/medical_records/_form.html.erb:
```diff
@@ -1,19 +1,24 @@
     <%= form.label :patient_id, style: "display: block" %>
-    <%= form.text_field :patient_id %>
+    <p>
+     <%= medical_record.patient_id %>
```
```suggestion
      <%= medical_record.patient_id %>
```

Ligator commented on app/controllers/medical_records_controller.rb:
```diff
@@ -60,12 +59,12 @@ def destroy
+      params.require(:medical_record).permit(:patient_id, :patology, :habits, :current_state, :notes)
     end
-end
+  end
```
All files should be finished with an empty line.
```suggestion
  end
  
```

NadiaArroyo25 committed:
Update app/views/medical_records/_medical_record.html.erb

Co-authored-by: Abel Pacheco <abelpacheco@outlook.com>

NadiaArroyo25 committed:
Update app/controllers/medical_records_controller.rb

Co-authored-by: Abel Pacheco <abelpacheco@outlook.com>

NadiaArroyo25 committed:
Update medical_records_controller.rb

NadiaArroyo25 committed:
actualize suggestions
